### PR TITLE
Add new component detection types and map to SBOMPackage

### PR DESCRIPTION
**Pull request by ksigmund**

### CODE REVIEW

File: src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/SpdxComponentExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Sbom.Contracts;
+using Microsoft.Sbom.Contracts.Enums;
+
+namespace Microsoft.Sbom.Adapters.ComponentDetection
+{
+    /// <summary>
+    /// Extensions methods for <see cref="SpdxComponent"/>.
+    /// </summary>
+    internal static class SpdxComponentExtensions
```
Comment (edgarrs) [inline src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/SpdxComponentExtensions.cs:13]: I thought we already have a mapping for SPDXComponent, isnt't that the case?

File: src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/DockerReferenceComponentExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        public static SBOMPackage? ToSbomPackage(this DockerReferenceComponent dockerReferenceComponent) => new ()
+        {
+            Id = dockerReferenceComponent.Id,
+            PackageUrl = dockerReferenceComponent.PackageUrl?.ToString(),
+            PackageName = dockerReferenceComponent.Digest,
+            Checksum = new[]
+            {
+                new Checksum
+                {
+                    Algorithm = AlgorithmName.SHA256,
```
Comment (edgarrs) [inline src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/DockerReferenceComponentExtensions.cs:28]: Is it always SHA256?

Comment (ksigmund) [reply src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/DockerReferenceComponentExtensions.cs:28]: I actually have that question myself.  I put that in there as a placeholder and am trying to find out where this is officially defined.  I can't quite recall how we knew which algorithm(s) to use for the other types that have checksums.

File: src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/SpdxComponentExtensions.cs (hunk 1 above)
Comment (ksigmund) [reply src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/SpdxComponentExtensions.cs:13]: I can't find any implementation of that, though there is functionality to convert and SpdxComponent to an external reference.  https://github.com/microsoft/sbom-tool/blob/40367fc11a6778d5cbb7da90466a87086ad4be9c/src/Microsoft.Sbom.Api/Converters/ComponentToExternalReferenceInfoConverter.cs#L62

File: src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/DockerReferenceComponentExtensions.cs (hunk 2 above)
Comment (ksigmund) [reply src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/DockerReferenceComponentExtensions.cs:28]: I found in the CD source code that SHA1 is used here.

**Review by aasim — APPROVED** [review]

**Pull request merged**